### PR TITLE
Reduce the docker image size

### PR DESCRIPTION
**Pull request by kramasamy**
- Using debian slim image from openjdk
- With slim image the size has reduced to 584 MB.

### CODE REVIEW

File: docker/dist/Dockerfile.dist.debian8 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# run heron tools install
+RUN /heron/heron-tools-install.sh && \
+    tar --strip-components=1 -m -zxvf /heron/heron-core.tar.gz -C /heron && \
+    rm -rf /heron/heron-core.tar.gz && \
+    rm -rf /heron/heron-tools-install.sh && \
+    rm -rf /opt/heron/heron-core/lib/scheduler/heron-local-scheduler.jar && \
+    rm -rf /opt/heron/heron-core/lib/scheduler/heron-mesos-scheduler.jar && \
+    rm -rf /opt/heron/heron-core/lib/scheduler/heron-slurm-scheduler.jar
+
+ENV HERON_HOME /opt/heron/heron-core/
```
Comment (cckellogg) [inline docker/dist/Dockerfile.dist.debian8]: this should be ENV HERON_HOME /heron/heron-core/

File: docker/dist/Dockerfile.dist.ubuntu14.04 [hunk 2]
```diff
@@ -18,14 +19,13 @@ ADD artifacts /heron
...
 
-RUN rm -f /heron/heron-tools-install.sh
-RUN rm -f /heron/heron-client-install.sh
-RUN rm -f /heron/heron-core.tar.gz
+RUN rm -f /heron/heron-tools-install.sh && \
+    rm -f /heron/heron-client-install.sh && \
+    rm -f /heron/heron-core.tar.gz
 
 ENV HERON_HOME /heron/heron-core/
 RUN export HERON_HOME
```
Comment (cckellogg) [inline docker/dist/Dockerfile.dist.ubuntu14.04]: we can remove line 31

File: docker/compile/Dockerfile.debian8 [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      libtool-bin \
+      python \
+      python2.7-dev \
+      software-properties-common \
+      python-setuptools \
+      zip \
+      unzip \
+      wget
+
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64
```
Comment (nwangtw) [inline docker/compile/Dockerfile.debian8]: suggest to move this line to beginning. not a blocker.

File: docker/scripts/build-docker.sh [hunk 4]
```diff
@@ -77,7 +76,7 @@ case $# in
     echo "  "
     echo "Usage: $0 <platform> <version_string> <artifact-directory> "
     echo "  "
-    echo "Platforms Supported: ubuntu14.04, ubuntu15.10, ubuntu16.04, centos7"
+    echo "Platforms Supported: debian8 ubuntu14.04, ubuntu15.10, ubuntu16.04, centos7"
```
Comment (nwangtw) [inline docker/scripts/build-docker.sh]: suggest to be the same order as build-artifacts.sh

File: docker/compile/Dockerfile.debian8 (hunk 3 above)
Comment (kramasamy) [reply docker/compile/Dockerfile.debian8]: removed this since the image already sets JAVA_HOME.

**Pull request merged**